src/languages/swift.js

```diff
@@ -275,6 +283,7 @@ export default function(hljs) {
     end: />/,
     keywords: KEYWORDS,
     contains: [
+      ...COMMENTS,
       ...KEYWORD_MODES,
       ...ATTRIBUTES,
       OPERATOR_GUARD,
```

Comment:
Can you think of any good reason the compiler shouldn't just flatten `contains` for you avoiding keeping track of which items and rules vs arrays of rules?

Reply:
I'm not sure what you mean here. Am I not already flattening the nested arrays with the leading `...` operator?

Reply:
Yes, but it's error prone and you have to constantly remember which is a mode and which is a list of modes... vs just not requiring the `...` and letting the compiler do the aggregation.

Reply:
Obviously `...` is more explicit, but is there value in that explicitness or does it just get in the way?

Reply:
What would the alternative code be?

This may be just a case of me not being a JS developer.

Reply:
```js
    contains: [
      ...COMMENTS,
      ...KEYWORD_MODES,
      ...ATTRIBUTES,
      OPERATOR_GUARD,
```

vs

The compiler figure it out for you.

```
    contains: [
      COMMENTS,
      KEYWORD_MODES,
      ATTRIBUTES,
      OPERATOR_GUARD,
```

The question is does forcing the reader/writer to keep track of what needs ... and what doesn't add any value... I'm not sure it does.

Reply:
I'm not interested in an array here ever having any special type of semantic meaning... so that means technically it would be easy to move this type of thing into the compiler and make the type passed (array or item) irrelevant.

Reply:
By "compiler", do you mean some part of HLJS that processes these language files? Or is this a JS "feature" ? In the first case, I'd still prefer the explicit version I think. It's less surprising/confusing.

For example, in https://github.com/highlightjs/highlight.js/pull/2938, I have the following code:

```js
keywords: [
  ...Swift.precedencegroupKeywords,
  ...Swift.literals
].join(' '),
```

I tried leaving off the leading `...` here, but that didn't work. Of course, this being JS and all, I didn't get any errors. It just didn't work 🤷‍♂️